mlflow/pipelines/steps/predict.py

```diff
@@ -0,0 +1,153 @@
+from mlflow.pipelines.cards import BaseCard
+from mlflow.pipelines.step import BaseStep
+from mlflow.pipelines.utils.execution import get_step_output_path
+from mlflow.pipelines.utils.step import get_pandas_data_profile
+from mlflow.utils._spark_utils import _get_active_spark_session
+
+_logger = logging.getLogger(__name__)
+
+
+# This should maybe imported from the preprocessing step for consistency
+_INPUT_FILE_NAME = "dataset.parquet"
+_SCORED_OUTPUT_FILE_NAME = "scored.parquet"
+
+
+class PredictStep(BaseStep):
+    def __init__(self, step_config: Dict[str, Any], pipeline_root: str):
+        super().__init__(step_config, pipeline_root)
+
+        self.run_end_time = None
+        self.execution_duration = None
+
+    def _build_profiles_and_card(self, scored_df) -> BaseCard:
+        # Build profiles for input dataset, and train / validation / test splits
+        scored_data_profile = get_pandas_data_profile(
+            scored_df.reset_index(drop=True),
+            "Profile of Scored Dataset",
+        )
+
+        # Build card
+        card = BaseCard(self.pipeline_name, self.name)
+        # Tab #1: data profile for scored data:
+        card.add_tab("Scored Data Profile", "{{PROFILE}}").add_pandas_profile(
+            "PROFILE", scored_data_profile
+        )
+        # Tab #2: run summary.
+        (
+            card.add_tab(
+                "Run Summary",
+                """
+                {{ SCORED_DATA_NUM_ROWS }}
+                {{ EXE_DURATION}}
+                {{ LAST_UPDATE_TIME }}
+                """,
+            ).add_markdown(
+                "SCORED_DATA_NUM_ROWS",
+                f"**Number of scored dataset rows:** `{len(scored_df)}`",
+            )
+        )
+
+        return card
+
+    def _run(self, output_directory):
+        from pyspark.sql.functions import struct
+
+        run_start_time = time.time()
+
+        # Get or create spark session
+        try:
+            spark = _get_active_spark_session()
+        except Exception as e:
+            raise MlflowException(
+                message=(
+                    "Encountered an error while searching for an active Spark session to"
+                    " score dataset with spark UDF. Please create a Spark session and try again."
+                ),
+                error_code=BAD_REQUEST,
+            ) from e
+        if not spark:
+            raise MlflowException(
+                message=(
+                    "No active SparkSession detected to score dataset with spark UDF. "
+                    "Please create a Spark session and try again."
+                ),
+                error_code=BAD_REQUEST,
+            )
+
+        # read cleaned dataset
+        ingested_data_path = get_step_output_path(
+            pipeline_root_path=self.pipeline_root,
+            step_name="ingest_scoring",
+            relative_path=_INPUT_FILE_NAME,
+        )
+        input_sdf = spark.read.parquet(ingested_data_path)
```

Comment:
Emit a warning at least here if it contains `prediction` column, indicating that the column will be overwritten in output.